stdlib/tkinter/__init__.pyi

```diff
@@ -3166,12 +3196,45 @@ class Text(Widget, XView, YView):
     def tag_ranges(self, tagName: str) -> tuple[_tkinter.Tcl_Obj, ...]: ...
     # tag_remove and tag_delete are different
     def tag_remove(self, tagName: str, index1: _TextIndex, index2: _TextIndex | None = None) -> None: ...
-    # TODO: window_* methods
-    def window_cget(self, index, option): ...
-    def window_configure(self, index, cnf: Incomplete | None = None, **kw): ...
+    @overload
+    def window_cget(self, index: _TextIndex, option: Literal["padx", "pady"]) -> int: ...
+    @overload
+    def window_cget(self, index: _TextIndex, option: Literal["stretch"]) -> Literal[0, 1]: ...
+    @overload
+    def window_cget(self, index: _TextIndex, option: Literal["align"]) -> _TextChildAlign: ...
+    @overload  # window is set to a widget, but read as the string name.
```

Comment:
We should also support setting it to a string, so that you can set any option to whatever `window_cget()` returns.

Pretty much all arguments can be given as strings in tkinter, but we type things with `... | str` only when passing in a string makes sense in practice. I think that is the case here.